packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutThreshold.tsx

```diff
@@ -449,24 +459,25 @@ export const ChartDonutThreshold: React.FunctionComponent<ChartDonutThresholdPro
   const renderChildren = () =>
     React.Children.toArray(children).map(child => {
       if (child.props) {
-        const datum = getData([{ ...child.props.data }]);
-        const orientation = child.props.donutOrientation || donutOrientation;
+        const {data: childData, ...childProps} = child.props;
+        const datum = Data.formatData([childData], childProps, ['x', 'y']); // Format child data independently of this component's props
+        const orientation = childProps.donutOrientation || donutOrientation;
         const dynamicTheme =
-          child.props.theme ||
-          getDonutThresholdDynamicTheme(child.props.themeColor || themeColor,
-            child.props.themeVariant || themeVariant);
+          childProps.theme ||
+          getDonutThresholdDynamicTheme(childProps.themeColor || themeColor,
+            childProps.themeVariant || themeVariant);
         return React.cloneElement(child, {
-          donutDx: child.props.donutDx || getDynamicDonutDx(dynamicTheme, orientation),
-          donutDy: child.props.donutDy || getDynamicDonutDy(dynamicTheme),
-          donutHeight: child.props.donutHeight || donutHeight - (theme.pie.height - dynamicTheme.pie.height),
+          donutDx: childProps.donutDx || getDynamicDonutDx(dynamicTheme, orientation),
+          donutDy: childProps.donutDy || getDynamicDonutDy(dynamicTheme),
+          donutHeight: childProps.donutHeight || donutHeight - (theme.pie.height - dynamicTheme.pie.height),
           donutOrientation: orientation,
-          donutWidth: child.props.donutWidth || donutWidth - (theme.pie.width - dynamicTheme.pie.width),
-          endAngle: child.props.endAngle || 360 * (datum[0]._y ? datum[0]._y / 100 : 100),
-          height: child.props.height || height,
-          showStatic: child.props.showStatic || false,
+          donutWidth: childProps.donutWidth || donutWidth - (theme.pie.width - dynamicTheme.pie.width),
+          endAngle: childProps.endAngle || 360 * (datum[0]._y ? datum[0]._y / 100 : 100),
+          height: childProps.height || height,
+          showStatic: childProps.showStatic || false,
           standalone: false,
           theme: dynamicTheme,
-          width: child.props.width || width
+          width: childProps.width || width
```

Comment:
Could we apply `... childProps` at the end? Thinking it would help make the code more readable.

Reply:
Yeah, I think we can do something like this:
```jsx
return React.cloneElement(child, {
  donutDx: getDynamicDonutDx(dynamicTheme, orientation),
  donutDy: getDynamicDonutDy(dynamicTheme),
  donutHeight: donutHeight - (theme.pie.height - dynamicTheme.pie.height),
  donutOrientation: orientation,
  donutWidth: donutWidth - (theme.pie.width - dynamicTheme.pie.width),
  endAngle: 360 * (datum[0]._y ? datum[0]._y / 100 : 100),
  height,
  showStatic: false,
  standalone: false,
  theme: dynamicTheme,
  width: width,
  ...childProps,
});
```